File: src/main/java/com/cxxcxx/zinecraft/core/entity/ModBlockEntity.java
package com.cxxcxx.zinecraft.core.entity;

import com.cxxcxx.zinecraft.core.Zinecraft;
import com.cxxcxx.zinecraft.core.block.ModBlock;
import net.minecraft.world.level.block.entity.BlockEntityType;

import java.util.function.Supplier;

public final class ModBlockEntity {
  private ModBlockEntity() {
  }  public static final Supplier<BlockEntityType<ExampleBlockEntity>> EXAMPLE_BLOCK_ENTITY = Zinecraft.BLOCK_ENTITIES.register(
      "example_block_entity", ExampleBlockEntity::new, ModBlock.EXAMPLE_ENTITY_BLOCK
  );

  public static void bootstrap() {
  }


}

File: src/main/kotlin/com/cxxcxx/zinecraft/core/entity/ModBlockEntity.kt
package com.cxxcxx.zinecraft.core.entity


import com.cxxcxx.zinecraft.core.Zinecraft
import com.cxxcxx.zinecraft.core.block.ModBlock
import net.minecraft.world.level.block.entity.BlockEntityType


object ModBlockEntity {

  val EXAMPLE_BLOCK_ENTITY: BlockEntityType<ExampleBlockEntity> =
    Zinecraft.BLOCK_ENTITIES.register(
      "example_block_entity", ::ExampleBlockEntity,
      ModBlock.EXAMPLE_ENTITY_BLOCK
    )
}
